simon8491/src/main/java/workshop/UserEdit.java

```diff
@@ -0,0 +1,38 @@
+package workshop;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+
+@WebServlet("/user/edit")
+public class UserEdit extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String id = req.getParameter("id");
+        UserDao userDao = new UserDao();
+        User user = userDao.readUser(Integer.parseInt(id));
+        req.setAttribute("user", user);
+        getServletContext().getRequestDispatcher("/users/edit.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        User user = new User();
+        UserDao userDao = new UserDao();
+        user.setId(Integer.parseInt(req.getParameter("id")));
+        user.setUserName(req.getParameter("userName"));
+        user.setEmail(req.getParameter("email"));
+        // password re-hash
+        String password = req.getParameter("password");
+        if (password != null && !password.isEmpty()) {
+            userDao.hashPassword(password);
```

Comment:
BUG: zaheshovane heslo si nikam neukladas...

```suggestion
user.setPassword(userDao.hashPassword(password));
```